### PR TITLE
ref(alerts): Rename Issue Rules and Metric Rules consistently

### PR DESCRIPTION
**Pull request by taylangocmen**
This renames and reorganizes most of the alerts folder:
- renamed `IncidentRule` and variants to `MetricRule`, similar to `IssueRule`, 
- and also changes the `issueRuleEditor` and  directory to `issueRules` to maintain consistency.
- moved `issueRuleEditor/` and `incidentRules/` to `rules/issue/` and `rules/metric/`
- moved `rules/details/` and `/details/` to `rules/metric/details/`  and `rules/issue/details` respectively
- moved `rules/metric/details.tsx` to `rules/metric/edit.tsx` because of name conflicts with the folder
- moved some utility function files under `util/`
- moved alerts list page components to `alerts/list/incidents/` and `alerts/list/rules`

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -82,8 +82,8 @@ build-utils/        @getsentry/owners-js-build
 /static/app/views/settings/projectAlerts/                       @getsentry/workflow
 /tests/js/spec/views/settings/projectAlerts/                    @getsentry/workflow
 
-/static/app/views/settings/incidentRules/                       @getsentry/workflow
-/tests/js/spec/views/settings/incidentRules/                    @getsentry/workflow
+/static/app/views/settings/metricRules/                         @getsentry/workflow
+/tests/js/spec/views/settings/metricRules/                      @getsentry/workflow
```
Comment (scttcper) [inline .github/CODEOWNERS:86]: i don't think this path exists

Comment (taylangocmen) [reply .github/CODEOWNERS:86]: Must be an old path, will delete

File: static/app/views/settings/components/settingsHeader.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import space from 'sentry/styles/space';
 // This is required to offer components that sit between this settings header
 // and i.e. dropdowns, some zIndex layer room
 //
-// e.g. app/views/settings/incidentRules/triggers/chart/
```
Comment (scttcper) [inline static/app/views/settings/components/settingsHeader.tsx:8]: meh

Comment (taylangocmen) [reply static/app/views/settings/components/settingsHeader.tsx:8]: might as well 🤷

File: static/app/views/settings/account/accountSecurity/index.tsx [hunk 3]
```diff
@@ -91,7 +91,7 @@ class AccountSecurity extends AsyncView<Props> {
               <ListLink to={recreateRoute('', this.props)} index>
                 {t('Settings')}
               </ListLink>
-              <ListLink to={recreateRoute('session-incidents/', this.props)}>
```
Comment (scttcper) [inline static/app/views/settings/account/accountSecurity/index.tsx]: revert change

**Review by scttcper — APPROVED** [review]

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**